File: MultiLanguageSandbox/src/test/deps/Visual Basic/46.vb
Module Module1
Function ReverseThreeDigitNumber(ByVal n As Integer) As Integer
    ' Extract each digit of the three-digit number
    Dim hundreds As Integer = n \ 100
    Dim tens As Integer = (n \ 10) Mod 10
    Dim units As Integer = n Mod 10
    
    ' Reverse the digits and form the new number
    Dim reversedNumber As Integer = units * 100 + tens * 10 + hundreds
    
    Return reversedNumber
End Function

    Sub Main()
        ' Test the ReverseThreeDigitNumber function
        Debug.Assert(ReverseThreeDigitNumber(358) = 853)
        Debug.Assert(ReverseThreeDigitNumber(100) = 1)
        Debug.Assert(ReverseThreeDigitNumber(250) = 52)
        Debug.Assert(ReverseThreeDigitNumber(999) = 999)
        Debug.Assert(ReverseThreeDigitNumber(123) = 321)
        Debug.Assert(ReverseThreeDigitNumber(505) = 505)
        Debug.Assert(ReverseThreeDigitNumber(210) = 12)
    End Sub
End Module

File: MultiLanguageSandbox/src/test/deps/perl/46.pl

sub reverse_three_digit_number {
    my ($number) = @_;
    # Convert the number to a string to handle each digit individually
    my $str = "$number";
    # Extract each digit
    my $hundreds = substr($str, 0, 1);
    my $tens = substr($str, 1, 1);
    my $units = substr($str, 2, 1);
    # Reverse the digits and return as a string
    return $units . $tens . $hundreds;
}


sub check {
    my $func = shift;
    die "Test failed! Expected '853', got '" . &$func(358) . "'" unless &$func(358) eq '853';
    die "Test failed! Expected '001', got '" . &$func(100) . "'" unless &$func(100) eq '001';
    die "Test failed! Expected '876', got '" . &$func(678) . "'" unless &$func(678) eq '876';
}

check(\&reverse_three_digit_number);